### PR TITLE
#39 QuestionDisplayPage 

### PR DESCRIPTION
**Pull request by jetcanfly**
A simple question display page.
Future plan:
1. add more tabs.
2. add more option like: login, logout, personal profile, so on to navigation bar.

### CODE REVIEW

File: SA3/qa_web/static/css/qa.css [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  vertical-align: middle;
+}
+
+tag {
+  background-color:#E0E0E0;
+  color:#500000;
+}
+
+
+#search {
```
Comment (13rinda) [inline SA3/qa_web/static/css/qa.css]: the search is not being used here now so it can be deleted.

File: SA3/qa_web/static/css/qa.css [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  -ms-transition: all 0.5s ease-in-out;
+  transition: all 0.5s ease-in-out;
+
+  -webkit-transform: translate(0px, -100%) scale(0, 0);
+  -moz-transform: translate(0px, -100%) scale(0, 0);
+  -o-transform: translate(0px, -100%) scale(0, 0);
+  -ms-transform: translate(0px, -100%) scale(0, 0);
+  transform: translate(0px, -100%) scale(0, 0);
+
+  opacity: 0;
```
Comment (13rinda) [inline SA3/qa_web/static/css/qa.css]: opacity?

File: SA3/qa_web/static/css/qa.css [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    border: 1px solid #bebab0;
+}
+.topbar {
+  width:100%;
+  height:7px;
+  position: absolute;
+  z-index:1000;
+
+  background: #9fa6ff;
+}
```
Comment (13rinda) [inline SA3/qa_web/static/css/qa.css]: screen shot 2018-02-16 at 11 09 54 am

Comment (13rinda) [reply SA3/qa_web/static/css/qa.css]: Not sure what is meant by line 168, as it is viewed as an odd icon.

File: SA3/qa_web/views.py [hunk 4]
```diff
@@ -147,3 +155,178 @@ def vote(request):
...
+        context['total_question_num'] = Questions.objects.count()
+        context['total_answer_num'] = Answers.objects.count()
+
+        #  try to page the question page
+        context['is_paginated'] = True  # whether the paginator is needed. It's a must.
+        all_questions_qs = Questions.objects.order_by('-creation_date')\
+            .select_related('owner')\
+            .annotate(num_answers=Count('answers', distinct=True),
+                      num_question_comments=Count('comments',
+                      distinct=True))
```
Comment (alvyn279) [inline SA3/qa_web/views.py]: Cool implementation of the count of answers to question and comments to question through `Count`.

File: SA3/qa_web/views.py [hunk 5]
```diff
@@ -147,3 +155,178 @@ def vote(request):
...
+        context = {
+            'left': left,
+            'right': right,
+            'left_has_more': left_has_more,
+            'right_has_more': right_has_more,
+            'first': first,
+            'last': last,
+        }
+
+        return context
```
Comment (13rinda) [inline SA3/qa_web/views.py]: same as line 168 the odd icon appears here also

File: SA3/qa_web/static/css/qa.css (hunk 3 above)
Comment (jetcanfly) [reply SA3/qa_web/static/css/qa.css]: @13rinda It seems it is a good style to always put the newline as a last character. I added one. Thanks.

File: SA3/qa_web/templates/qa_web/base_navigator.html [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+</ul>
+{% endif %}
+{% block content %}
+
+{% endblock content %}
+ 
+</div>
+<hr>
+<center>
+  <a href="http://www.djangoproject.com/"><green>Django</green></a> ·
```
Comment (Lercerss) [inline SA3/qa_web/templates/qa_web/base_navigator.html]: Good thing to add these references to the footer.

**Comment by jetcanfly** [comment]
@13rinda Hi Brinda, I've made some changes according to your advice. If it looks good to you maybe we could merge this to the master. Thanks~

**Pull request merged**